src/components/inputs/TextField/TextField.tsx

```diff
@@ -259,6 +264,7 @@ const TextField: React.FC<TextFieldProps> = ({
         className: classes.label,
         ...InputLabelProps
       }}
+      helperText={helperText || ' '}
```

Comment:
This is not ok, it will generate even more height inconsistency between components. It must be removed.
What you can do, you can reduce the height of the error message, so it won't be such a big discrepancy (see the comment in the TextField theme overrides).